src/main/java/org/syftkog/web/test/framework/Driver.java

```diff
@@ -0,0 +1,664 @@
+   * @param executorService
+   */
+  public Driver(WebDriver driver, Capabilities driverCapabilities, ExecutorService executorService) {
+    this(driver, driverCapabilities, executorService, null);
+  }
+
+  /**
+   *
+   * @return
+   */
+  @Override
+  public WebDriver getWrappedDriver() {
+    return driver;
+  }
+
+  /**
+   *
+   * @param text
+   */
+  public void logStep(String text) {
+    stepLogger.log(text);
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Capabilities getCapabilities() {
+    return driverCapabilities;
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Page getPreviousPage() {
+    return pageHistory.lastElement();
+  }
+
+  /**
+   *
+   * @param page
+   * @return
+   */
+  public Driver addPageToDriverHistory(Page page) {
+    pageHistory.push(page);
+    return this;
+  }
+
+  /**
+   *
+   * @param url
+   */
+  @Override
+  public void get(String url) {
+    urlHistory.add(url);
+    driver.get(url);
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Page navigateBack() {
+    logStep("Navigate BACK using browser.");
+    driver.navigate().back();
+
+    if (!pageHistory.empty()) {
+      pageHistory.pop();
+      if (!pageHistory.empty()) {
+        Page lastPage = pageHistory.peek();
+        return lastPage;
+      } else {
+        LOG.debug("Cannot return previous page if it is not already stored in the driver. Did you click and navigate to a new page without pushing it to the page history?");
+        return null;
+      }
+    }
+    return null;
+  }
+
+  /**
+   *
+   * @return
+   */
+  @Override
+  public String getCurrentUrl() {
+    return driver.getCurrentUrl();
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Driver prepareToNavigateToNewUrl() {
+    urlHistory.push(getCurrentUrl());
+    return this;
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Driver waitForNavigateToNewUrl() {
+    return waitUntilUrlChange(urlHistory.peek(), implicitWaitTimeInSeconds);
+  }
+
+  /**
+   *
+   * @return
+   */
+  @Override
+  public String getTitle() {
+    return driver.getTitle();
+  }
+
+  /**
+   *
+   * @param by
+   * @return
+   */
+  @Override
+  public List<WebElement> findElements(By by) {
+    return driver.findElements(by);
+  }
+
+  /**
+   *
+   * @param by
+   * @return
+   */
+  @Override
+  public WebElement findElement(By by) {
+    return driver.findElement(by);
+  }
+
+  /**
+   *
+   * @return
+   */
+  @Override
+  public String getPageSource() {
+    return driver.getPageSource();
+  }
+
+  private synchronized void terminate() {
+    Assert.assertTrue(inUse == false);
+    if (driver != null) {
+      try {
+        driver.quit();
+        if (factory != null) {
+          factory.reportThatDriverQuitSuccessfully(this);
+        }
+        driver = null; // set the driver null so we don't try to use it anymore.
+      } catch (UnsupportedCommandException ex) {
+        LOG.error(ex.toString());
+      }
+    }
+
+  }
+
+  // Closes a single browser window.

```

Comment:
Is there any reason not to put this comment into the JavaDoc block following this line?